### PR TITLE
Windows-related fixes and GHA for Windows.

### LINKED ISSUE #892: Add GitHub Actions for Windows to ensure RNP builds on Windows/MinGW
Since it now builds on MinGW, it's probably time we add ~~AppVeyor~~ GHA testing... 👏

(Thanks to @kaie !)

_Originally posted by @ronaldtse in https://github.com/riboseinc/rnp/pull/883#issuecomment-515192850_

### LINKED ISSUE #961: mingw "make install" should install the .dll file
After running "make install", the destination area doesn't contain librnp-0.dll

Also, it installs some files which probably shouldn't get copied, like cmake files, see below.

Only the following are installed:
```
-- Install configuration: "Release"
-- Installing: /home/user/local-botan-mingw-32/lib/liblibrnp-0.dll.a
-- Installing: /home/user/local-botan-mingw-32/include/rnp-0/rnp.h
-- Installing: /home/user/local-botan-mingw-32/lib/cmake/rnp/rnp-targets.cmake
-- Installing: /home/user/local-botan-mingw-32/lib/cmake/rnp/rnp-targets-release.cmake
-- Installing: /home/user/local-botan-mingw-32/lib/cmake/rnp/rnp-config.cmake
-- Installing: /home/user/local-botan-mingw-32/lib/cmake/rnp/rnp-config-version.cmake
-- Installing: /home/user/local-botan-mingw-32/lib/pkgconfig/librnp-0.pc
-- Installing: /home/user/local-botan-mingw-32/bin/rnp.exe
-- Installing: /home/user/local-botan-mingw-32/bin/rnpkeys.exe
```

### PR DESCRIPTION
**Pull request by ni4**
This PR finally adds working GHA for Windows.
However, currently it disables ruby-rnp test - it will be fixed back as a separate PR.
Also I had a lot of problems with building Json-C/Botan/GnuPG from sources so now those are installed as binaries via MSYS's pacman.

Closes #892 
As a side effect it fixes #961

### CODE REVIEW

**Comment by ni4** [comment]
@ronaldtse @dewyatt Could you please review this one?

File: ci/main.sh [hunk 1]
```diff
@@ -25,19 +25,30 @@ cmakeopts=(
 [ "$BUILD_MODE" = "coverage" ] && cmakeopts+=("-DENABLE_COVERAGE=yes")
 [ "$BUILD_MODE" = "sanitize" ] && cmakeopts+=("-DENABLE_SANITIZERS=yes")
 
+if [[ "$(get_os)" = "msys" ]]; then
+  cmakeopts+=("-G")
+  cmakeopts+=("MSYS Makefiles")
```
Comment (dewyatt) [inline ci/main.sh]: I think you can do this in one line: `cmakeopts+=("-G" "MSYS Makefiles")`

Comment (ronaldtse) [reply ci/main.sh]: @ni4 feel free to fix this and merge anytime. Thanks!

Comment (ni4) [reply ci/main.sh]: @ronaldtse done!

**Pull request merged**